commands/commands.go

```diff
@@ -346,6 +347,24 @@ func buildProgressMeter(dryRun bool) *progress.ProgressMeter {
 	)
 }
 
+// findLocks finds matching locks using the given "lc" *locking.Client, and all
+// other parameters for the lookup. If an error was encountered, it will be
+// returned immediately, otherise a map of lock.Path -> lock will be returned
+// instead.
+func findLocks(lc *locking.Client, filter map[string]string, limit int, localOnly bool) (map[string]locking.Lock, error) {
+	locks, err := lc.SearchLocks(filter, limit, localOnly)
```

Comment:
Thanks for brining this up 😄 . I went ahead and changed the implementation of the `Search()` method on the `*locking.Client` to only try and deserialize the response if it was a `200 OK`. If the response was a 404 or 410, the lack of deserialization would return an empty slice of locks, making the lock-related parts of the pre-push command effectively a no-op.

I don't think any changes are needed for the `Lock()` and `Unlock()` methods, since those are mutative operations and _should_ error out if the server doesn't implement them.

Changes are in: 6fb728a.